src/main/java/com/google/cloud/resourcemanager/ResourceManagerImpl.java

```diff
@@ -273,6 +280,261 @@ public Policy replacePolicy(final String projectId, final Policy newPolicy) {
+                  EXCEPTION_HANDLER,
+                  serviceOptions.getClock());
+      String cursor = result.x();
+      Iterable<Constraint> constraints =
+          result.y() == null
+              ? ImmutableList.<Constraint>of()
+              : Iterables.transform(
+                  result.y(),
+                  new Function<
+                      com.google.api.services.cloudresourcemanager.model.Constraint, Constraint>() {
+                    @Override
+                    public Constraint apply(
+                        com.google.api.services.cloudresourcemanager.model.Constraint
+                            constraintPb) {
+                      return new Constraint(
+                          serviceOptions.getService(),
+                          new ConstraintInfo.BuilderImpl(ConstraintInfo.fromPb(constraintPb)));
+                    }
+                  });
+      return new PageImpl<>(
+          new ConstraintPageFetcher(resource, request, serviceOptions, cursor, optionsMap),
+          cursor,
+          constraints);
+    } catch (RetryHelperException ex) {
+      throw ResourceManagerException.translateAndThrow(ex);
+    }
+  }
+
+  private static class OrgPolicyPageFetcher implements NextPageFetcher<OrgPolicy> {
+
+    private static final long serialVersionUID = 2158209410430566961L;
+    private final String resource;
+    private final ListOrgPoliciesRequest request;
+    private final Map<ResourceManagerRpc.Option, ?> requestOptions;
+    private final ResourceManagerOptions serviceOptions;
+
+    OrgPolicyPageFetcher(
+        String resource,
+        ListOrgPoliciesRequest request,
+        ResourceManagerOptions serviceOptions,
+        String cursor,
+        Map<ResourceManagerRpc.Option, ?> optionMap) {
+      this.resource = resource;
+      this.request = request;
+      this.requestOptions =
+          PageImpl.nextRequestOptions(ResourceManagerRpc.Option.PAGE_TOKEN, cursor, optionMap);
+      this.serviceOptions = serviceOptions;
+    }
+
+    @Override
+    public Page<OrgPolicy> getNextPage() {
+      return listOrgPolicies(resource, request, serviceOptions, requestOptions);
+    }
+  }
+
+  @Override
+  public Page<OrgPolicy> listOrgPolicies(
+      final String resource, final ListOrgPoliciesRequest request) {
+    Map<ResourceManagerRpc.Option, Object> optionsMap = new HashMap<>();
+    optionsMap.put(ResourceManagerRpc.Option.PAGE_TOKEN, request.getPageToken());
+    optionsMap.put(ResourceManagerRpc.Option.PAGE_SIZE, request.getPageSize());
+    return listOrgPolicies(resource, request, getOptions(), optionsMap);
+  }
+
+  private static Page<OrgPolicy> listOrgPolicies(
+      final String resource,
+      final ListOrgPoliciesRequest request,
+      final ResourceManagerOptions serviceOptions,
+      final Map<ResourceManagerRpc.Option, ?> optionsMap) {
+    try {
+      Tuple<String, Iterable<com.google.api.services.cloudresourcemanager.model.OrgPolicy>> result =
```

Comment:
This is too hard to read, between the line breaks and the fully package qualified names. I can't figure out what this is supposed to or if it's correct.

Reply:
Already removed the use of Tuple